frame/elections-phragmen/src/lib.rs

```diff
@@ -116,6 +124,131 @@ type BalanceOf<T> =
 type NegativeImbalanceOf<T> =
 	<<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::NegativeImbalance;
 
+/// Helper functions for migrations of this module.
+pub mod migrations {
+	use super::*;
+	use frame_support::{migration::StorageKeyIterator, Twox64Concat};
+
+	/// Migrate from the old legacy voting bond (fixed) to the new one (per-vote dynamic).
+	///
+	/// Will only be triggered if storage version is V1.
+	pub fn migrate_voters_to_recorded_deposit<T: Trait>(old_deposit: BalanceOf<T>) -> Weight {
+		if <Module<T>>::pallet_storage_version() == StorageVersion::V1 {
+			let mut count = 0;
+			<StorageKeyIterator<T::AccountId, (BalanceOf<T>, Vec<T::AccountId>), Twox64Concat>>::new(
+				<Voting<T>>::module_prefix(),
+				b"Voting",
+			)
+			.for_each(|(who, (stake, votes))| {
+				// Insert a new value into the same location, thus no need to do `.drain()`.
+				let deposit = old_deposit;
+				let voter = Voter { votes, stake, deposit };
+				<Voting<T>>::insert(who, voter);
+				count += 1;
+			});
+
+			PalletStorageVersion::put(StorageVersion::V2RecordedDeposit);
```

Comment:
is there a reason for splitting all the migrations ? also the doc should state that this writes V2 to storage as other migration can't be triggered after this one as passed.

Reply:
This part is not done yet, I want to use the crate version stuff.